loggers/zap/zap.go

```diff
@@ -0,0 +1,107 @@
+//Package zap provides a BaseLogger implementation for uber/zap
+package zap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-coldbrew/log/loggers"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type logger struct {
+	logger *zap.Logger
+	opt    loggers.Options
+	cfg    zap.Config
+}
+
+func (l *logger) Log(ctx context.Context, level loggers.Level, skip int, args ...interface{}) {
+
+	msg := fmt.Sprint(args...)
```

Comment:
should we add support for sugared and non sugared logger ?

Reply:
yes, we should let the application choose the type of zap logger they want, but the difference would be we won't be able to log data with the same reason of sugared logger, i.e to have logger without any overhead on identifying the types.

Reply:
we could add this as a loggers.Option right ?

Reply:
yes, setting up this logger is easy, however, using it bit difficult. sugared logger is quite straightforward, non sugared logger expects typed fields, we pass `map[string]interface{}` to `log.Info`, which then we would have to convert by using type assertion which defeats the purpose of using the zap.